ballerina/graphql_client.bal

```diff
@@ -0,0 +1,107 @@
+import ballerina/http;
+ 
+# The [Ballerina](https://ballerina.io/) generic client for GraphQL(https://graphql.org/) APIs.
+public isolated client class Client {
+    final http:Client httpClient;
+
+    # Gets invoked to initialize the `connector`.
+    #
+    # + serviceUrl - URL of the target service
+    # + clientConfig - The configurations to be used when initializing the `connector`
+    # + return - An error at the failure of client initialization
+    public isolated function init(string serviceUrl, http:ClientConfiguration clientConfig = {})  returns ClientError? {
+        http:Client|http:ClientError httpClient = new (serviceUrl, clientConfig);
+        if httpClient is http:ClientError {
+             return error RequestError("GraphQL Client Error", httpClient);
+        }
+        self.httpClient = httpClient;
+    }
+
+    # Executes a GraphQL query and data binds the GraphQL response to a record with data and extensions 
+    # which is a subtype of GenericResponse. The errors of the GraphQL response are directed in the Ballerina error path.
+    #
+    # + query - The GraphQL query. For example `query countryByCode($code:ID!) {country(code:$code) {name}}`.
+    # + variables - The GraphQL variables. For example `{"code": "<variable_value>"}`.
+    # + headers - The GraphQL API headers to execute each query 
+    # + targetType - The payload, which is expected to be returned after data binding. For example 
+    #                `type CountryByCodeResponse record {| map<json?> __extensions?; record {| record{|string name;|}? country; |} data;`
+    # + return - The payload (if the `targetType` is configured) or a `graphql:ClientError` if failed to execute the query
+    remote isolated function executeWithType(string query, map<anydata>? variables = (), 
+                                             map<string|string[]>? headers = (), 
+                                             typedesc<GenericResponse|record{}|json> targetType = <>) 
+                                             returns targetType|ClientError = @java:Method {
+        'class: "io.ballerina.stdlib.graphql.runtime.client.QueryExecutor",
+        name: "executeWithType"
+    } external;
+
+    private isolated function processExecuteWithType(typedesc<GenericResponse|record{}|json> targetType, 
+                                                     string query, map<anydata>? variables, 
+                                                     map<string|string[]>? headers) 
+                                                     returns GenericResponse|record{}|json|ClientError {
+        http:Request request = new;
+        json graphqlPayload = getGraphqlPayload(query, variables);
+        request.setPayload(graphqlPayload);
+        json|http:ClientError httpResponse = self.httpClient->post("", request, headers = headers);
+
+        if httpResponse is http:ClientError {
+            check handleHttpClientErrorResponse(httpResponse);
```

Comment:
```suggestion
            return handleHttpClientErrorResponse(httpResponse);
```

Reply:
The change is applied in [47e5a38](https://github.com/ballerina-platform/module-ballerina-graphql/pull/663/commits/47e5a38537e0e31ee452d51e9943f81c773e3a95). Thank you.